src/app/state/blogpost/blogpost.effects.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { map, mergeMap } from 'rxjs/operators';
+import { Post } from 'src/app/blogpost/post';
+import { BlogpostService } from 'src/app/blogpost/blogpost.service';
+
+/* NgRx */
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { PostApiActions, PostPageActions } from './actions';
+
+@Injectable()
+export class BlogpostEffects {
+  constructor(
+    private actions$: Actions,
+    private blogpostService: BlogpostService
+  ) {}
+
+  loadPosts$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(PostPageActions.loadPosts),
+      mergeMap(() =>

```

Comment:
it's better to use switchMap for load related action and mergeMap for save related action.

you can see the difference by calling dispatch twice:

this.store.dispatch(PostPageActions.loadPosts());
this.store.dispatch(PostPageActions.loadPosts());